src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -14,144 +15,152 @@ import './ContentTagsDropDownSelector.scss';
 import { useTaxonomyTagsData } from './data/apiHooks';
 
 const ContentTagsDropDownSelector = ({
-  taxonomyId, level, subTagsUrl, lineage, tagsTree,
+  taxonomyId, level, lineage, tagsTree, searchTerm,
 }) => {
   const intl = useIntl();
+
   // This object represents the states of the dropdowns on this level
   // The keys represent the index of the dropdown with
   // the value true (open) false (closed)
   const [dropdownStates, setDropdownStates] = useState({});
+  const isOpen = (tagValue) => dropdownStates[tagValue];
 
-  const [tags, setTags] = useState([]);
-  const [nextPage, setNextPage] = useState(null);
+  const [numPages, setNumPages] = useState(1);
+  const parentTagValue = lineage.length ? decodeURIComponent(lineage[lineage.length - 1]) : null;
+  const { hasMorePages, tagPages } = useTaxonomyTagsData(taxonomyId, parentTagValue, numPages, searchTerm);
 
-  // `fetchUrl` is initially `subTagsUrl` to fetch the initial data,
-  // however if it is null that means it is the root, and the apiHooks
-  // would automatically handle it. Later this url is set to the next
-  // page of results (if any)
-  //
-  // TODO: In the future we may need to refactor this to keep track
-  // of the count for how many times the user clicked on "load more" then
-  // use useQueries to load all the pages based on that.
-  const [fetchUrl, setFetchUrl] = useState(subTagsUrl);
+  const [prevSearchTerm, setPrevSearchTerm] = useState(searchTerm);
 
-  const isOpen = (i) => dropdownStates[i];
+  // Reset the page and tags state when search term changes
+  // and store search term to compare
+  if (prevSearchTerm !== searchTerm) {
+    setPrevSearchTerm(searchTerm);
+    setNumPages(1);
+  }
```

Comment:
```suggestion
  useEffect(() => {
    if (prevSearchTerm !== searchTerm) {
      setPrevSearchTerm(searchTerm);
      setNumPages(1);
    }
  }, [prevSearchTerm, searchTerm]);
```

Reply:
Hmm, I think it's actually better without `useEffect`. The React docs explicitly advise against `useEffect` in this case: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes Or am I missing something?

Reply:
@bradenmacdonald You are correct. I did not know that this is recommended practice in this case. I just noticed a state change during render and that set off a red flag.